core/trino-main/src/main/java/io/trino/execution/StageStateMachine.java

```diff
@@ -535,7 +534,7 @@ public StageInfo getStageInfo(Supplier<Iterable<TaskInfo>> taskInfosSupplier)
             inputBlockedTime += taskStats.getInputBlockedTime().roundTo(NANOSECONDS);
 
             bufferedDataSize += taskInfo.getOutputBuffers().getTotalBufferedBytes();
-            outputBufferUtilization = Optionals.combine(outputBufferUtilization, taskInfo.getOutputBuffers().getUtilization(), TDigestHistogram::mergeWith);
+            taskInfo.getOutputBuffers().getUtilization().ifPresent(bufferUtilizationHistograms::add);
```

Comment:
Will this not increase peak memory usage, as we will have for a short period of time more TDigestHistograms in memory?

Reply:
yes, that is the trade-off of bulk processing vs two at a time processing, I don't think this requires a lot of memory or will stay in memory for long.